### DISCUSSION TITLE
how to use regexp parameters in rewrites destination

### DISCUSSION THREAD

**Discussion by Capricair** [vercel/next.js#40935] [Help] [answered]
like this:
source: "/(aaa.txt|bbb.xml)",
destination: "/$1"

**Comment by icyJoseph** [comment, marked as answer]
Hi,

As [per docs](https://nextjs.org/docs/api-reference/next.config.js/redirects#regex-path-matching), matched parameters can be used in the destination as follows:

```js
/** @type {import('next').NextConfig} */
const nextConfig = {
  reactStrictMode: true,
  async redirects() {
    return [
      {
        source: "/post/:slug(bar.xml|foo.txt)",
        destination: "/news/:slug", // Matched parameters can be used in the destination
        permanent: false,
      },
    ];
  },
};

module.exports = nextConfig;
```

## Edit

Btw, the matchers here are more of a path-based-regexp-matcher, that in turn creates a proper Regexp, and they can allow for custom path matching Regular expressions within, see https://github.com/pillarjs/path-to-regexp

**Reply by Capricair** [reply]
It's very useful, thank you very much!
